pydantic/env_settings.py

```diff
@@ -46,41 +45,34 @@ def _build_values(
         _secrets_dir: Union[Path, str, None] = None,
     ) -> Dict[str, Any]:
         return deep_update(
-            self._build_secrets(_secrets_dir), self._build_environ(_env_file, _env_file_encoding), init_kwargs
+            self._build_secrets_files(_secrets_dir), self._build_environ(_env_file, _env_file_encoding), init_kwargs
         )
 
-    def _build_secrets(self, _secrets_dir: Union[Path, str, None] = None) -> Dict[str, Optional[str]]:
+    def _build_secrets_files(self, _secrets_dir: Union[Path, str, None] = None) -> Dict[str, Optional[str]]:
         """
-        Build secrets suitable for passing to the Model.
+        Build fields from "secrets" files.
         """
         secrets: Dict[str, Optional[str]] = {}
 
         secrets_dir = _secrets_dir or self.__config__.secrets_dir
         if secrets_dir is None:
             return secrets
 
-        secrets_path = Path(secrets_dir)
+        secrets_path = Path(secrets_dir).expanduser()
```

Comment:
Is this documented?

Reply:
No, but it's pretty obvious

Reply:
Related https://github.com/samuelcolvin/pydantic/pull/1804